### PR TITLE
IDEX-3087: added predefined recipes

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok

File: assembly-sdk-war/src/main/java/org/eclipse/che/api/deploy/ApiModule.java [hunk 1]
```diff
@@ -52,6 +53,8 @@ protected void configure() {
         bind(WorkspaceService.class);
         bind(MachineClient.class).to(MachineClientImpl.class).in(Singleton.class);
 
+        bind(RecipeLoader.class).in(Singleton.class);
```
Comment (garagatyi) [inline assembly-sdk-war/src/main/java/org/eclipse/che/api/deploy/ApiModule.java]: It is singleton already, why you explicitly add annotation again?

Comment (akorneta) [inline assembly-sdk-war/src/main/java/org/eclipse/che/api/deploy/ApiModule.java]: +1 my fall

**Comment by vparfonov** [comment]
ok

File: assembly-sdk-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties [hunk 2]
```diff
@@ -65,4 +65,6 @@ machine.server.terminal.archive=${catalina.base}/web-terminal/terminal
 machine.default_mem_size_mb=1024
 machine.supported_docker_version=1.6.0,1.6.1,1.6.2,1.7.1,1.8.1
 
-git.server.uri.prefix=git
\ No newline at end of file
+git.server.uri.prefix=git
+
+codenvy.local.infrastructure.recipes=predefined-recipes.json
```
Comment (mshaposhnik) [inline assembly-sdk-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: Maybe codenvy.local.infrastructure.recipes.filename (or file) ?

Comment (skabashnyuk) [inline assembly-sdk-war/src/main/webapp/WEB-INF/classes/codenvy/che.properties]: che.local.infrastructure.recipes

**Pull request merged**